packages/core-sdk/resolvers/account-resolvers/interface/query.graphql

```diff
@@ -1,72 +1,130 @@
 #import { Token } into TokenResolver from "ens/interface.token-resolvers.defiwrapper.eth"
 
 type Query {
-  getTokenBalances(address: String!): [TokenBalance!]!
+  getTokenBalances(accountAddress: String!): TokenBalancesList!
 
-  # TODO
-  # getTokenTransfers(address: String!): TransfersList!
+  getTokenTransfers(
+    accountAddress: String!
+    tokenAddress: String!
+    vsCurrency: String!
+    options: Options
+  ): TransfersList!
 
-  # TODO
-  # getTransactions(address: String!): TransfersList!
+  getTransactions(accountAddress: String!, vsCurrency: String!, options: Options): TransactionsList!
+}
+
+type TokenBalancesList {
+  account: String!
+  chainId: String!
+  tokenBalances: [TokenBalance!]!
 }
 
 type TokenBalance {
   token: TokenResolver_Token!
   balance: String!
 }
 
-# TODO
-# enum TransferType {
-#   IN,
-#   OUT,
-# }
-
-# type Transfer {
-#   txHash: String!
-#   from: String!
-#   to: String!
-#   value: String!  # BigDecimal
-#   transferType: TransferType!
-#   timestamp: String!
-# }
-
-# type TransfersList {
-#   token: Token!
-#   transfers: [Transfer!]!
-#   pagination: Pagination!
-# }
-
-# type Options {
-#   pagination: PaginationOptions
-#   blockRange: BlockRangeOptions
-# }
-
-# type PaginationOptions {
-#   page: Int!
-#   perPage: Int!
-# }
-
-# type BlockRangeOptions {
-#   startBlock: Int  # default: 0
-#   endBlock: Int  # default: latest
-# }
-
-# type Pagination {
-#   total: Int!
-#   perPage: Int!
-#   page: Int!
-#   hasMore: Boolean!
-# }
-
-# type Transaction {
-#   hash: String!
-#   from: String!
-#   to: String!
-#   value: String!  # BigDecimal
-#   gasUsed: Int!  # Gas used for the transaction
-#   gasLimit: Int!  # Gas offered for the transaction
-#   gasPrice: String!  # Gas price in gwei (BigDecimal)
-#   timestamp: String!  # Timestamp of the transaction
-#   blockHeight: Int!  # Block height of the transaction
-#   offset: Int!
-# }
+type Options {
+  pagination: PaginationOptions
+  blockRange: BlockRangeOptions
+}
+
+type PaginationOptions {
+  page: Int!
+  perPage: Int!
+}
+
+type BlockRangeOptions {
+  startBlock: Int # default: 0
+  endBlock: Int # default: latest
+}
+
+type Pagination {
+  total: Int
+  perPage: Int
+  page: Int
+  hasMore: Boolean
+}
+
+type GasInfo {
+  offered: String! # Gas offered for the transaction
```

Comment:
should `offered` be renamed to `limit`?

Reply:
I don't mind but I kept it `offered` since it's more readable let me know if you think `limit` would be the better choice of word due to conventions.

Reply:
I think `offered` is fine. Is that what is used in the API?

Reply:
Yeah, that's why I kept it the same and it made sense to me.